backend/studytimeboard/routes.py

```diff
@@ -148,15 +148,32 @@ def api_handle_go_event():
 @app.route("/api/hold", methods=["POST"])
 def api_handle_hold_event():
     data = json.loads(request.data)
-    username = data["username"]
+    username = data[USERNAME]
     if username in dbapi.all_users():
         date = datetime.now(TZ)
         end_time = datetime2time(datetime.now(TZ))
         dbapi.into_hold(username, date, end_time)
         return {"status": "success"}, 200
     else:
         return {"status": "error", "message": FlashMessages.NO_SUCH_USER}, 400
+    
+@app.route("/api/duration", methods=["POST"])
```

Comment:
I would change this web api's name from duration to interval. In frontend both terms are used, where duration usually refers to how long the user has studied e.g. "20 minutes", and interval usually refers to the time range e.g. "8:00-10:00". Here user is clearly logging a new interval instead of a new duration